src/main/java/org/encryptDecryptMachine/EncryptionMachine.java

```diff
@@ -0,0 +1,114 @@
+
+    /**
+     * The main method of the program. It handles user interaction, encryption of the key,
+     * and encryption of the message.
+     *
+     * @param args Command-line arguments (not used in this program).
+     */
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+
+        printIntro();
+
+        encryptAndPrintKey(scanner);
+
+        int numWords = getNumWordsFromUserInput(scanner);
+
+        getAndEncryptWords(numWords, scanner);
+
+        System.out.println("\nMessage fully encrypted. Happy secret messaging!");
+    }
+
+    /**
+     * Encrypts a single letter using the Caesar cipher.
+     *
+     * @param letter The letter to encrypt.
+     * @return The encrypted letter.
+     */
+    private static char encryptLetter(char letter) {
+        int index = ALPHABET.indexOf(letter);
+        int newIndex = (index + SHIFT) % ALPHABET.length();
+        return ALPHABET.charAt(newIndex);
+    }
+
+    /**
+     * Encrypts a word by encrypting each of its letters using the Caesar cipher.
+     *
+     * @param word The word to encrypt.
+     * @return The encrypted word.
+     */
+    private static String encryptWord(String word) {
+        StringBuilder encryptedWord = new StringBuilder();
+        for (int i = 0; i < word.length(); i++) {
+            char encryptedLetter = encryptLetter(word.charAt(i));
+            encryptedWord.append(encryptedLetter);
+        }
+        return encryptedWord.toString();
+    }
+
+    /**
+     * Intro text for the encryption machine user input
+     */
+    private static void printIntro() {
+        System.out.println("Welcome to the CSCI717 Encryption Machine Construction");
+        System.out.println("The program lets you encrypt a message");
+        System.out.println("with a key for your recipient to decrypt!\n");
+    }
+
+    /**
+     * Gets the key from user, encrypts it, and prints it.
+     *
+     * @param scanner The Scanner object used for input
+     */
+    private static void encryptAndPrintKey(Scanner scanner) {
+        System.out.print("Enter key: ");
+        String key = scanner.next();
+        String encryptedKey = encryptWord(key);
+        System.out.println("\"" + key + "\" has been encrypted to: " + encryptedKey + "\n");
+    }
+
+    /**
+     * Gets the number of words to be encrypted from user input.
+     *
+     * @param scanner The Scanner object used for input
+     */
+    private static int getNumWordsFromUserInput(Scanner scanner) {
+        System.out.print("How many words is your message?: ");
+        int numWords = scanner.nextInt();
+        scanner.nextLine();
+        return numWords;
+    }
+
+    /**
+     * Takes user input for each word, encrypts it, and prints it.
+     *
+     * @param numWords How many words to encrypt
+     * @param scanner The Scanner object used for input
+     */
+    private static void getAndEncryptWords(int numWords, Scanner scanner) {
```

Comment:
All of your methods are also private. This makes them hard to test, as a test project is often separate and the methods that they test have to be public in order for them to be read.

Reply:
Also, if I were to test this method, I'd have to make or Mock a Scanner object inside the test method, which is not great, so I'd recommend having it removed as well.

Reply:
> All of your methods are also private
Fair point, I usually only like exposing a single public method (not counting getters/setters) to a class which then gets called for unit testing. It can be argued though that the methods of this class may be used on their own other than in this class.
Would probably need to be restructured as we add more functionality.

Will make these methods public though!

Reply:
> Mock a Scanner object inside the test metho

Passing the Scanner as a parameter was noted as a requirement in the instructions:
<img width="838" alt="image" src="https://github.com/user-attachments/assets/ac5f4ff9-778b-4e49-bc64-73765cd1007e">

Scanner is pretty simple and rather than mocking it (with something like Mockito), it would probably be fine to do something like this to simulate user input in the unit tests:

```
        String input = "secret\n3\nhello\nworld\nagain\n";
        InputStream in = new ByteArrayInputStream(input.getBytes());
        Scanner scanner = new Scanner(in);
```

I think this is going to be needed for the Black box testing assignment as well, as were supposed to test the code as if we didn't know the underlying code (by just calling main()).

That said, capturing output and simulating input is definitely a pain though!